models/repo_collaboration.go

```diff
@@ -186,9 +186,46 @@ func (repo *Repository) DeleteCollaboration(uid int64) (err error) {
 		return err
 	}
 
+	// Unassign a user from any issue (s)he has been assigned to in the
+	// repository
+	if err := repo.removeIssueAssignees(sess, uid); err != nil {
+		return err
+	}
+
 	return sess.Commit()
 }
 
+func (repo *Repository) removeIssueAssignees(e Engine, userID int64) error {
+
+	err := repo.GetOwner()
+	if err != nil {
+		// Very weird case
+		return fmt.Errorf("removeIssueAssignees: Unable to load owner for repo: %d Error: %v", repo.ID, err)
+	}
+
+	perm, err := getUserRepoPermission(e, repo, repo.Owner)
```

Comment:
you test if RepoOwner has access but this function should check if **userID** has access!

Reply:
Yikes, missed that.. Updated